configmap/informed_watcher.go

```diff
@@ -61,24 +62,61 @@ type InformedWatcher struct {
 	informer corev1informers.ConfigMapInformer
 	started  bool
 
+	// cfgs are the default ConfigMaps to use if the real ones do not exist or are deleted.
+	cfgs map[string]*corev1.ConfigMap
+
 	// Embedding this struct allows us to reuse the logic
 	// of registering and notifying observers. This simplifies the
-	// InformedWatcher to just setting up the Kubernetes informer
+	// InformedWatcher to just setting up the Kubernetes informer.
 	ManualWatcher
 }
 
 // Asserts that InformedWatcher implements Watcher.
 var _ Watcher = (*InformedWatcher)(nil)
 
-// Start implements Watcher
+// Asserts that InformedWatcher implements DefaultingWatcher.
+var _ DefaultingWatcher = (*InformedWatcher)(nil)
+
+// WatchWithDefault implements DefaultingWatcher.
+func (i *InformedWatcher) WatchWithDefault(cm corev1.ConfigMap, o Observer) {
+	if i.cfgs == nil {
```

Comment:
Move this to `NewInformedWatcher`?

Reply:
I decided against that because people can (but probably shouldn't) make an `InformedWatcher` directly, without going through `NewInformedWatcher`, so I would want this check here regardless.

Reply:
How? `sif` and `informer` are private and cannot be set outside of the provided constructors.

Reply:
You're right, people can do it, but it will panic as soon as someone calls `Start()` because `informer` is `nil`. So panicking in a slightly different spot isn't worse.

Moved.